sycl/include/CL/sycl/group.hpp

```diff
@@ -153,11 +263,14 @@ template <int Dimensions = 1> class group {
   //    Get a linearized version of the work-group id. Calculating a linear
   //    work-group id from a multi-dimensional index follows the equation 4.3.
   template <int dims = Dimensions>
-  typename detail::enable_if_t<(dims == 3), size_t> get_linear_id() const {
+  typename detail::enable_if_t<(dims == 3), size_t>
+  get_group_linear_id() const {
     return (index[0] * groupRange[1] * groupRange[2]) +
            (index[1] * groupRange[2]) + index[2];
   }
 
+  bool leader() const { return false; }
```

Comment:
The specification for this function says:
> Return true for exactly one work-item in the [work-group](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#work-group), if the calling work-item is the leader of the work-group, and false for all other work-items in the work-group.
>
> The leader of the work-group is determined during construction of the work-group, and is invariant for the lifetime of the work-group. The leader of the work-group is guaranteed to be the work-item with a local id of 0.

Because of the last sentence, I think the correct implementation is:
```c++
bool leader() const {
  return (get_local_linear_id() == 0);
}
```

Reply:
Done